### ISSUE TITLE
Epic: Support controller screens on the Traktor S4 MK3

### ISSUE THREAD

**Issue by ywwg** [mixxxdj/mixxx#13004]
### Feature Description

This is an epic bug to track work for supporting external screens on controllers, with an initial implementation for the Tratktor S4 MK3

Task List:

- [ ] #3967 
- [ ] #11407 
- [ ] #12199 
- [ ] #12558 
- [ ] #12559 
- [ ] #12557

**Comment by ywwg** [comment]
@acolombier

**Comment by ywwg** [comment]
I'll keep myself assigned because mixxx helped pay for the controller I'm using to do the testing and work

**Comment by acolombier** [comment]
Another task to add:
- [ ] #13011 (required for testing, fixing tests)

**Comment by ywwg** [comment]
@acolombier can you write an update here about what we have included now, and what is left to do?  I believe the big issue right now is the controller data sharing which needs design work.  (I am so happy with how quickly this is coming together, I assumed screen support would be a year of work!)

**Comment by acolombier** [comment]
> I believe the big issue right now is the controller data sharing which needs design work

Yes, that's correct. I think we are waiting for a formal design proposal on how to make that doable with the current engine, without closing any doors for the future of controllers. 

As it stands, if you merge both branches from #12199 and #12557, everything appears to be working just fine. 

> I assumed screen support would be a year of work!

Technically, I've been working on that feature since march 2023, but I know what you mean. I'm also glad we managed to get that fat PR out of the way in a fairly short time.

**Comment by acolombier** [comment]
@JoergAtGithub do you have any knowledge on how to do the [libwdi](https://github.com/pbatard/libwdi) intergration? Just looking at [examples](https://github.com/pbatard/libwdi/tree/master/examples) but not quite sure what is needed.

If you have any WIP or POC branch, I could use some help :)

**Comment by JoergAtGithub** [comment]
I also investigated about libwdi. But I never used it. I think the first step should be to register the driver with the prebuilt and signed Zadig tool. Once this works, I would try to transfer these settings 1:1 to libwdi.
